src/storage/src/source/persist_source.rs

```diff
@@ -190,3 +194,218 @@ where
 
     (ok_stream, err_stream, token)
 }
+
+pub fn persist_source_listen_split<G>(
+    scope: &G,
+    persist_clients: Arc<Mutex<PersistClientCache>>,
+    metadata: CollectionMetadata,
+    as_of: Antichain<Timestamp>,
+) -> (
+    Stream<G, (Row, Timestamp, Diff)>,
+    Stream<G, (DataflowError, Timestamp, Diff)>,
+    Rc<dyn Any>,
+)
+where
+    G: Scope<Timestamp = mz_repr::Timestamp>,
+{
+    let worker_index = scope.index();
+    let peers = scope.peers();
+
+    let data_shard = metadata.data_shard.clone();
+
+    let persist_clients_moveable = persist_clients.clone();
```

Comment:
nit: oh because we have two operators now? I'd probably name these according to the operator that moves them or something else a bit more descriptive

it might also "just work" to shadow them with a clone with the same name, move it, and then move the original one into the second operator? (i.e. just put a `let persist_clients = persist_clients.clone();` here)

Reply:
Came up with clearer names